From 41faf8288a96147095051947c38401adc4fc2a17 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 25 Sep 2020 08:28:53 -0500
Subject: [PATCH 54/54] arm64: dts: beacon-imx8mn-spidev: Add SPIdev dts file
 for Nano

There is a custom device tree with spidev for the i.Mx8M Mini, but
nothing existed for Nano.  This patch introduces a comparable device
tree for the i.Mx8MN Nano beacon board which disables the EEPROM driver
to allow the SPIDEV driver to read/write to the EEPROM and it also
removes UART3, and changes the pinmux to support ECSPI1 which is
routed to header, J80.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../dts/freescale/beacon-imx8mn-spidev.dts    | 61 +++++++++++++++++++
 1 file changed, 61 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/beacon-imx8mn-spidev.dts

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-spidev.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mn-spidev.dts
new file mode 100644
index 000000000000..284d731ca244
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-spidev.dts
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2020 Beacon EmbeddedWorks
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8mn.dtsi"
+#include "beacon-imx8mn-som.dtsi"
+#include "beacon-imx8mn-baseboard.dtsi"
+#include "beacon-imx8mn-lvds.dtsi"
+
+/ {
+	model = "beacon EmbeddedWorks i.MX8M Nano Development Kit";
+	compatible = "fsl,imx8mn";
+
+	chosen {
+		stdout-path = &uart2;
+	};
+
+	aliases {
+		/delete-property/ serial2;
+	};
+};
+
+/* UART3 shares pins with ECSPI1, so they cannot both exist */
+/delete-node/ &uart3;
+
+&ecspi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	status = "okay";
+	cs-gpios = <&gpio5 9 GPIO_ACTIVE_LOW>;
+
+	spidev1@0x00 {
+               compatible = "spidev";
+               spi-max-frequency = <5000000>;
+               reg = <0>;
+       };
+};
+
+&ecspi2 {
+	/delete-node/ at25@0;
+	
+	spidev1@0x00 {
+               compatible = "spidev";
+               spi-max-frequency = <5000000>;
+               reg = <0>;
+       };
+};
+
+&iomuxc {
+	pinctrl_ecspi1: uart3grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_ECSPI1_SCLK_ECSPI1_SCLK		0x82
+			MX8MN_IOMUXC_ECSPI1_MOSI_ECSPI1_MOSI		0x82
+			MX8MN_IOMUXC_ECSPI1_MISO_ECSPI1_MISO		0x82
+			MX8MN_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
+		>;
+	};
+};
-- 
2.17.1

